floors/floors.go

```diff
@@ -0,0 +1,251 @@
+		*extFloorRules.Skipped = true
+		return []error{}
+	}
+
+	floorErrList = validateFloorRulesAndLowerValidRuleKey(modelGroup.Schema, modelGroup.Schema.Delimiter, modelGroup.Values)
+	if len(modelGroup.Values) > 0 {
+		for i, imp := range request.GetImp() {
+			desiredRuleKey := createRuleKey(modelGroup.Schema, request.BidRequest, request.Imp[i])
+			matchedRule, isRuleMatched := findRule(modelGroup.Values, modelGroup.Schema.Delimiter, desiredRuleKey, len(modelGroup.Schema.Fields))
+
+			floorVal = modelGroup.Default
+			if isRuleMatched {
+				floorVal = modelGroup.Values[matchedRule]
+			}
+
+			floorMinVal, floorCur, err := getMinFloorValue(extFloorRules, request.Imp[i], conversions)
+			if err == nil {
+				floorVal = math.Round(floorVal*10000) / 10000
+				bidFloor := floorVal
+				if floorMinVal > float64(0) && floorVal < floorMinVal {
+					bidFloor = floorMinVal
+				}
+
+				if bidFloor > float64(0) {
+					imp.BidFloor = math.Round(bidFloor*10000) / 10000
+					imp.BidFloorCur = floorCur
+				}
+				if isRuleMatched {
+					updateImpExtWithFloorDetails(imp, matchedRule, floorVal, imp.BidFloor)
+				}
+			} else {
+				floorErrList = append(floorErrList, fmt.Errorf("Error in getting FloorMin value : '%v'", err.Error()))
+			}
+		}
+		err := request.RebuildImp()
+		if err != nil {
+			return append(floorErrList, err)
+		}
+	}
+	return floorErrList
+}
+
+// isPriceFloorsDisabled check for floors are disabled at account or request level
+func isPriceFloorsDisabled(account config.Account, bidRequestWrapper *openrtb_ext.RequestWrapper) bool {
+	return isPriceFloorsDisabledForAccount(account) || isPriceFloorsDisabledForRequest(bidRequestWrapper)
+}
+
+// isPriceFloorsDisabledForAccount check for floors are disabled at account
+func isPriceFloorsDisabledForAccount(account config.Account) bool {
+	return !account.PriceFloors.Enabled
+}
+
+// isPriceFloorsDisabledForRequest check for floors are disabled at request
+func isPriceFloorsDisabledForRequest(bidRequestWrapper *openrtb_ext.RequestWrapper) bool {
+	requestExt, err := bidRequestWrapper.GetRequestExt()
+	if err == nil {
+		if prebidExt := requestExt.GetPrebid(); prebidExt != nil && prebidExt.Floors != nil && !prebidExt.Floors.GetEnabled() {
+			return true
+		}
+	}
+	return false
+}
+
+// resolveFloors does selection of floors fields from requet JSON and dynamic fetched floors JSON if dynamic fetch is enabled
+func resolveFloors(account config.Account, bidRequestWrapper *openrtb_ext.RequestWrapper, conversions currency.Conversions) (*openrtb_ext.PriceFloorRules, []error) {
+	var errlist []error
+	var floorsJson *openrtb_ext.PriceFloorRules
+
+	reqFloor := extractFloorsFromRequest(bidRequestWrapper)
+	if reqFloor != nil {
+		floorsJson, errlist = createFloorsFrom(reqFloor, openrtb_ext.FetchNone, openrtb_ext.RequestLocation)
+	} else {
+		floorsJson, errlist = createFloorsFrom(nil, openrtb_ext.FetchNone, openrtb_ext.NoDataLocation)
+	}
```

Comment:
This can just be `floorsJson, errlist = createFloorsFrom(nil, openrtb_ext.FetchNone, openrtb_ext.NoDataLocation)`.

Reply:
added `if` to pass `openrtb_ext.RequestLocation` when request is present, else `openrtb_ext.NoDataLocation`